### PR TITLE
starknet-rs version raised + Transaction rpc 0.4.0 adaptations

### PR DESCRIPTION
**Pull request by taco-paco**
## Checklist:

- [ ] Applied formatting - `./scripts/format.sh`
- [ ] No linter errors - `./scripts/clippy_check.sh`
- [ ] Performed code self-review
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Documented the changes
- [ ] Linked the issues which this PR resolves
- [ ] Updated the tests
- [ ] All tests are passing - `cargo test`

### CODE REVIEW

File: crates/starknet/src/starknet/add_deploy_account_transaction.rs [hunk 1]
```diff
@@ -118,7 +132,11 @@ mod tests {
         let (txn_hash, _) = starknet.add_deploy_account_transaction(transaction).unwrap();
         let txn = starknet.transactions.get_by_hash_mut(&txn_hash).unwrap();
 
-        assert_eq!(txn.status, TransactionStatus::Rejected);
+        assert_eq!(txn.finality_status, None);
+        assert_eq!(
+            txn.execution_result.revert_reason(),
+            Some(format!("Actual fee exceeds max fee. Actual: 3709, Max: {}", fee_raw).as_str())
```
Comment (ivpavici) [inline crates/starknet/src/starknet/add_deploy_account_transaction.rs:138]: are we sure about the actual fee always being 3709 ?

File: crates/starknet-server/tests/common/devnet.rs [hunk 2]
```diff
@@ -92,6 +91,7 @@ impl BackgroundDevnet {
                 .spawn()
                 .expect("Could not start background devnet");
 
+        let pid = process.id();
```
Comment (ivpavici) [inline crates/starknet-server/tests/common/devnet.rs:94]: where is this used?

File: crates/starknet-server/tests/get_transaction_receipt_by_hash.rs [hunk 3]
```diff
@@ -115,15 +115,15 @@ mod get_transaction_receipt_by_hash_integration_tests {
...
+        let result: MaybePendingTransactionReceipt = devnet
             .json_rpc_client
             .get_transaction_receipt(declare_transaction.transaction_hash)
             .await
             .unwrap();
 
         match result {
             MaybePendingTransactionReceipt::Receipt(TransactionReceipt::Declare(declare)) => {
-                assert_eq!(declare.status, TransactionStatus::Rejected);
+                assert_eq!(declare.execution_result.revert_reason(), Some("asdasd"));
```
Comment (ivpavici) [inline crates/starknet-server/tests/get_transaction_receipt_by_hash.rs:126]: why  "asdasd" :D

File: crates/types/src/rpc/transaction_receipt.rs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    pub transaction_hash: TransactionHash,
+    #[serde(flatten)]
+    pub output: TransactionOutput,
+    #[serde(flatten)]
+    pub execution_status: ExecutionResult,
+    #[serde(flatten)]
+    pub maybe_pending_properties: MaybePendingProperties,
+}
+
+// impl From<TransactionReceiptWithStatus> for MaybePendingTransactionReceipt {
```
Comment (ivpavici) [inline crates/types/src/rpc/transaction_receipt.rs:66]: leftover?

File: crates/starknet-server/tests/get_transaction_receipt_by_hash.rs (hunk 3 above)
Comment (taco-paco) [reply crates/starknet-server/tests/get_transaction_receipt_by_hash.rs]: My custom naming conventions. Removed

File: crates/starknet-server/tests/common/devnet.rs (hunk 2 above)
Comment (taco-paco) [reply crates/starknet-server/tests/common/devnet.rs]: Leftover from debugging. PID is used to know to which process I shall attach debugger

**Pull request closed without merging**